dataset_preprocessing/remove_duplicates.py

```diff
@@ -0,0 +1,75 @@
+from imutils import paths
+import numpy as np
+import argparse
+import cv2
+import os
+
+def dhash(image, hashSize=8):
+    # Convert the image to a single-channel grayscale image
+    if(image is not None):
+        # Resize the image according to the hashSize (Line 13). The algorithm requires that the width of the image have exactly 1 more column than the height as is evident by the dimension tuple.
+        gray = cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)
+
+        # Adding a single column (width) so we can compute the horizontal gradient.
+        # Resize the image according to the hashSize (Line 13). The algorithm requires that the width of the image have exactly 1 more column than the height as is evident by the dimension tuple.
+        resized = cv2.resize(gray, (hashSize + 1, hashSize))
+
+        # Compute the (relative) horizontal gradient between adjacent column pixels
+        diff = resized[:, 1:] > resized[:, :-1]
+
+        # Convert the difference image to a hash and return it
+        return sum([2 ** i for (i, v) in enumerate(diff.flatten()) if v])
+
+# construct the argument parser and parse the arguments
+ap = argparse.ArgumentParser()
+ap.add_argument("-d", "--dataset", required=True,
+	help="path to input dataset")
+ap.add_argument("-r", "--remove", type=int, default=-1,
+	help="whether or not duplicates should be removed (i.e., dry run)")
+
+args = vars(ap.parse_args())
+
+# grab the paths to all images in our input dataset directory and then initialize our hashes dictionary
+print("[INFO] computing image hashes...")
+imagePaths = list(paths.list_images(args["dataset"]))
+hashes = {}
+
+# loop over our image paths
+for imagePath in imagePaths:
+	# load the input image and compute the hash
+	image = cv2.imread(imagePath)
+	h = dhash(image)
+
+	# grab all image paths with that hash, add the current image path to it, and store the list back in the hashes dictionary
+	p = hashes.get(h, [])
+	p.append(imagePath)
+	hashes[h] = p
+
+# loop over the image hashes
+for (h, hashedPaths) in hashes.items():
+	# check to see if there is more than one image with the same hash
+    if len(hashedPaths) > 1:
+    	# check to see if this is a dry run
```

Comment:
Please get out of the habit of leaving things out of functions. All of these different small pieces should be bundled into atomic functions, then call them inside a `if __name__ == '__...`